relm-gen-widget/src/lib.rs

```diff
@@ -556,3 +607,115 @@ fn gen_set_child_prop_calls(widget: &Widget) -> Option<ImplItem> {
         None
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use syn::parse_expr;
+    use syn::{ExprKind};
+
+    #[test]
+    #[should_panic(expected = "Expected `view!` macro, found `foo` instead")]
+    fn incorrect_view_macro_name() {
+        let macro_text = "foo! {
+            gtk::Window {}
+        }";
+        let parsed_expr: ExprKind = parse_expr(macro_text).unwrap().node;
+        let mac = match parsed_expr {
+            ExprKind::Mac(mac) => mac,
+            _ => panic!("Expected ExprKind::Mac(mac), found {:#?}", parsed_expr),
+        };
+        let mut driver = Driver {
+            data_method: None,
+            generic_types: None,
+            model_type: None,
+            model_param_type: None,
+            msg_model_map: None,
+            msg_type: None,
+            other_methods: vec![],
+            properties_model_map: None,
+            root_method: None,
+            root_type: None,
+            root_widget: None,
+            root_widget_expr: None,
+            root_widget_type: None,
+            update_method: None,
+            view_macro: Some(mac),
+            widget_model_type: None,
+            widget_msg_type: None,
+            widget_parent_id: None,
+            widgets: HashMap::new(),
+        };
+        driver.view_validation_before_impl();
+    }
+
+    #[test]
+    #[should_panic(expected = "`view!` macro is empty, must contain one top-level item")]
+    fn empty_view_macro() {
+        let macro_text = "view! {
+        }";
+        let parsed_expr: ExprKind = parse_expr(macro_text).unwrap().node;
+        let mac = match parsed_expr {
+            ExprKind::Mac(mac) => mac,
+            _ => panic!("Expected ExprKind::Mac(mac), found {:#?}", parsed_expr),
+        };
+        let mut driver = Driver {
```

Comment:
Please call `Driver::new()` instead.

Reply:
Fixed. For some reason I was thinking I couldn't access the `view_macro` field because it wasn't marked `pub` 🤔